### PR TITLE
Added support for Ollama running locally or publicly hosted api

### PR DESCRIPTION
**Pull request by zaverichintan**
Added support for Ollama running locally or public hosted gpt Api.

Overview:
- Added adapter for supporting various LLM models in the backend.
- Adjusted main.py to read correct information from config and secret yaml files.
- Readme file updated with various parameters to be configured in data folder files and the use of the parameters

### CODE REVIEW

**Comment by feder-cr** [comment]
@zaverichintan we need a more scalable structure, I would like to have an adapter that can interact faaciously with different APIs

**Comment by ralyodio** [comment]
we need to be able to pass the url to the ollama api too, mine is on a server somewhere.

**Comment by zaverichintan** [comment]
Added support for various models by setting it up in the config file and also have endpoint for Ollama API

**Comment by feder-cr** [comment]
@zaverichintan tske inspiration by this: https://github.com/feder-cr/linkedIn_auto_jobs_applier_with_AI/pull/46

**Review by Aadik1ng — APPROVED** [review]

File: data_folder/config.yaml [hunk 1]
```diff
@@ -39,4 +39,8 @@ companyBlacklist:
 
 titleBlacklist:
   - word1
-  - word2
\ No newline at end of file
+  - word2
+
+llm_model_type: [openai / ollama]
+llm_model: ['gpt-4o' / 'mistral:v0.3']
+llm_api_url: [https://api.pawan.krd/cosmosrp/v1',  http://127.0.0.1:11434/]
```
Comment (barakplasma) [inline data_folder/config.yaml:46]: this yaml doesn't look right to me

File: src/gpt.py [hunk 2]
```diff
@@ -113,10 +114,19 @@ def parse_llmresult(self, llmresult: AIMessage) -> Dict[str, Dict]:
...
-            ChatOpenAI(model_name="gpt-4o-mini", openai_api_key=openai_api_key, temperature=0.4)
-        )
+    def __init__(self, config, llm_api_key):
+        llm_model_type = config['llm_model_type']
+        llm_model = config['llm_model']
+        llm_api_url = config['llm_api_url']
+
+        print('Using {0} with {1} from {2}'.format(llm_model_type, llm_model, llm_api_url))
+
+        if llm_model_type == "ollama":
```
Comment (barakplasma) [inline src/gpt.py]: I think you can remove this if by recommending that people use the ollama openai API compatible endpoint https://github.com/ollama/ollama/blob/main/docs/openai.md

also allows any other openai compatible provider to be used (there are many), also via something like https://docs.litellm.ai/docs/proxy/quick_start

File: data_folder/config.yaml (hunk 1 above)
Comment (zaverichintan) [reply data_folder/config.yaml]: This is for showing all the options - corrected the yaml file

File: src/gpt.py (hunk 2 above)
Comment (zaverichintan) [reply src/gpt.py]: This feature is experimental as mentioned on the site you provided:
`Ollama provides experimental compatibility with parts of the [OpenAI API](https://platform.openai.com/docs/api-reference) to help connect existing applications to Ollama`

**Comment by sammyview80** [comment]
reply = self.llm.invoke(messages)
AttributeError: 'AIAdapter' object has no attribute 'invoke'
This is the error occured while using claude sonnet 3.5.

**Comment by zaverichintan** [comment]
I will have to work on the pull request, kindly ignore for now from reviewing

**Comment by zaverichintan** [comment]
I corrected the Claude integration - kindly review and merge it

**Comment by feder-cr** [comment]
@zaverichintan 
Really, great job—you hit the nail on the head! I would just suggest a couple of adjustments:

Optional Fields: Please add optional fields in the YAML file only for experienced users. Ensure that the original user-side version remains unchanged.

Documentation: Create a separate section in the README specifically for changing the model. The configuration for normal users should remain the same.

(Consider making a tutorial on how to integrate Ollama.)

**Comment by zaverichintan** [comment]
@feder-cr I created a separate section in Readme and added guide to deploy ollama locally. Also config yaml file is adjusted.

**Pull request merged**